Library/Homebrew/cmd/deps.rb

```diff
@@ -72,10 +72,10 @@ def deps_for_formula(f, recursive = false)
 
     if recursive
       deps = f.recursive_dependencies do |dependent, dep|
-        Dependency.prune if ignores.any? { |ignore| dep.send(ignore) } && !includes.any? { |include| dep.send(include) } && !dependent.build.with?(dep)
+        Dependency.prune if ignores.any? { |ignore| dep.send(ignore) } && !includes.any? { |include| dep.send(include) } && (!dependent.build.with?(dep) || ignores.any? { |ignore| dep.send(ignore) })
```

Comment:
I think it would be good to break this into multiple lines/statements. It was unwieldy before and it's a bit crazy now.